app/src/main/java/uk/gov/onelogin/login/usecase/VerifyIdToken.kt

```diff
@@ -0,0 +1,86 @@
+package uk.gov.onelogin.login.usecase
+
+import android.util.Log
+import javax.inject.Inject
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import uk.gov.android.network.api.ApiRequest
+import uk.gov.android.network.api.ApiResponse
+import uk.gov.android.network.client.GenericHttpClient
+import uk.gov.onelogin.tokens.verifier.JwtVerifier
+
+interface VerifyIdToken {
+    operator fun invoke(
+        idToken: String,
+        jwksUrl: String,
+        callback: (Boolean) -> Unit
+    )
+}
+
+@Suppress("TooGenericExceptionCaught")
+class VerifyIdTokenImpl @Inject constructor(
+    private val httpClient: GenericHttpClient,
+    private val verifier: JwtVerifier
+) : VerifyIdToken {
+    override fun invoke(
+        idToken: String,
+        jwksUrl: String,
+        callback: (Boolean) -> Unit
+    ) {
+        MainScope().launch {
```

Comment:
shall we use `IO` for this? or the networking client does this automatically?

Reply:
yep, refactored this bit, shouldn't really be having a global scope being launched